crates/router/src/routes/app.rs

```diff
@@ -75,14 +91,41 @@ impl AppState {
             conf,
             #[cfg(feature = "email")]
             email_client,
+            #[cfg(feature = "kms")]
+            kms_secrets,
         }
     }
 
-    pub async fn new(conf: Settings, shut_down_signal: oneshot::Sender<()>) -> Self {
+    pub async fn new(conf: settings::Settings, shut_down_signal: oneshot::Sender<()>) -> Self {
         Self::with_storage(conf, StorageImpl::Postgresql, shut_down_signal).await
     }
 }
 
+#[allow(clippy::panic)]
+///
+/// # Panics
+/// panics of the function fails 2 times
+///
+pub async fn retry<F, T, E, Fut>(func: F, retry_count: u8) -> T
```

Comment:
What error are you trying to recover from? If network issues, an immediate retry won't be of any use.

Reply:
as @SanchithHegde suggested, I would have added a delay, but it's uncertain is how much time will the network recover. It would be better to just do it once as the server will not boot up, and we can simply redeploy.

Reply:
Yes Nishant, either put some delay like 1 second. Or give up after first try itself, since this is startup scenario. Anything while the server is up, we may have to think on handling failure/retries.